stl/inc/chrono

```diff
@@ -619,8 +618,61 @@ namespace chrono {
+            return _Whole + _Part;
+        }
+
+        _NODISCARD static time_point now() noexcept { // get current time
+#if (defined(_M_IX86) || defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)) && !defined(_M_CEE_PURE)
+            // Implement atomics avoiding <atomic> header dependency
+            static struct {
+                volatile long long _Freq        = LLONG_MAX;
+                volatile long long _Ctr_base    = LLONG_MAX;
+                volatile long long _Result_base = LLONG_MAX;
+            } _Cached;
+            const long long _Freq_from_cache = _Atomic_load_ll_relaxed(&_Cached._Freq);
+            const long long _Ctr_base        = _Atomic_load_ll_relaxed(&_Cached._Ctr_base);
+            const long long _Result_base     = _Atomic_load_ll_relaxed(&_Cached._Result_base);
+            if (_Freq_from_cache != LLONG_MAX && _Ctr_base != LLONG_MAX && _Result_base != LLONG_MAX) {
+                // Fast path
+                const long long _Ctr = _Query_perf_counter();
+                return time_point(duration((_Ctr - _Ctr_base) * period::den / _Freq_from_cache + _Result_base));
+            }
+            // Caclulate with two divisions to prevent overflow
+            const long long _Freq   = _Query_perf_frequency();
+            const long long _Ctr    = _Query_perf_counter();
+            const long long _Result = _Scale_large_counter(_Ctr, _Freq);
+            if (_Atomic_compare_exchange_strong_ll_cst_seq(&_Cached._Freq, _Freq, LLONG_MAX)) {
+                // This is the first result, save current result as base for fast path
+                _Atomic_compare_exchange_strong_ll_cst_seq(&_Cached._Ctr_base, _Ctr, LLONG_MAX);
+                _Atomic_compare_exchange_strong_ll_cst_seq(&_Cached._Result_base, _Result, LLONG_MAX);
+            }
+            // if _Result is not saved as first, it is still compatible with fast result
+            return time_point(duration(_Result));
+#else // ^^^ known hardware && !defined(_M_CEE_PURE) / unknown hardware || defined(_M_CEE_PURE) vvv
+            const long long _Freq = _Query_perf_frequency(); // doesn't change after system boot
+            const long long _Ctr  = _Query_perf_counter();
+            return time_point(duration(_Scale_large_counter(_Ctr, _Freq)));
+#endif // (defined(_M_IX86) || defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)) && !defined(_M_CEE_PURE)
+        }
+
+#if (defined(_M_IX86) || defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)) && !defined(_M_CEE_PURE)
+        static long long _Atomic_load_ll_relaxed(volatile long long* _Mem) {
+            // Copy from _Atomic_storage<_Ty, 8>::load
+#if defined(_M_IX86) || defined(_M_ARM64)
+            return __iso_volatile_load64(_Mem);
+#elif defined(_M_X64)
+            return *_Mem;
+#else // _M_ARM
+            return __ldrexd(_Mem);
+#endif // hardware
+        }
+
+        static bool _Atomic_compare_exchange_strong_ll_cst_seq(

```

Comment:
This function is named `cst_seq` but I believe it should be named `seq_cst` ("sequential consistency"), like everything in the Standard.